registry/storage/garbagecollect.go

```diff
@@ -114,8 +128,23 @@ func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 		return fmt.Errorf("failed to mark: %v", err)
 	}
 
+	// call GC extension handlers' mark
```

Comment:
tldr: the current way it's implemented, the first sweep doesn't mark any artifact manifests. It's only till the second pass that the artifact manifests are considered and the referrers are walked. then each artifact manifest and its blobs are added to the markset. 

The second mark step is necessary to mark all of the artifact manifests and their blobs that have a valid subject image that must be kept. The first pass (which has the existing functionality) will enumerate each manifest and check if it's untagged. If it's untagged and the GC has been set to clean up untagged, it will then add that manifest to the manifest Delete array. However, if the manifest in question is not eligible for deletion (`IsEligibleForDeletion` is used to determine this) then the current manifest will not be added to the delete list. The second mark step is needed to complete all artifact related operations which includes traversing the referrers. It goes through the manifests again and checks if the manifest should be deleted. If it does not need to be deleted, then it starts walking the referrers and recursively adds each manifest and its blobs to the markset. If does need to be deleted, then it recursively walks the referrers but this time it adds each artifact manifest to a delete set to be vacuumed away.